examples/benchmark.py

```diff
@@ -31,183 +124,469 @@ def downloadAmberSuite():
+parser.add_argument('--style', default='simple', dest='style', choices=STYLES, help=f'output style: {STYLES} [default: simple]')
+parser.add_argument('--outfile', default=None, dest='outfile', help='output filename for benchmark logging (must end with .yaml or .json)')
+parser.add_argument('--serialize', default=None, dest='serialize', help='if specified, output serialized test systems for Folding@home or other uses')
+parser.add_argument('--verbose', default=False, action='store_true', dest='verbose', help='if specified, print verbose output')
 args = parser.parse_args()
-if args.platform is None:
-    parser.error('No platform specified')
-print('Platform:', args.platform)
-if args.platform in ('CUDA', 'OpenCL'):
-    print('Precision:', args.precision)
-    if args.device is not None:
-        print('Device:', args.device)
-
-# Run the simulations.
-
-if args.test is None:
-    for test in ('gbsa', 'rf', 'pme', 'apoa1rf', 'apoa1pme', 'apoa1ljpme', 'amoebagk', 'amoebapme'):
+
+# Collect system information
+system_info = dict()
+import socket, platform
+system_info['hostname'] = socket.gethostname()
+system_info['timestamp'] = datetime.utcnow().isoformat()
+system_info['openmm_version'] = mm.version.version
+system_info['cpuinfo'] = cpuinfo()
+system_info['cpuarch'] = platform.processor()
+system_info['system'] = platform.system()
+# TODO: Capture information on how many CPU threads will be used
+
+# Attempt to get GPU info
+try:
+    import subprocess
+    cmd = 'nvidia-smi --query-gpu=driver_version,gpu_name --format=csv,noheader'
+    output = subprocess.check_output(cmd, shell=True, text=True)
+    system_info['nvidia_driver'], system_info['gpu'] = output.strip().split(', ')
+except Exception as e:
+    pass
+
+for key, value in system_info.items():
+    print(f'{key}: {value}')
+
+if args.outfile is not None:
+    # Remove output file
+    import os
+    if os.path.exists(args.outfile):
+        os.unlink(args.outfile)
+    # Write system info
+    appendTestResult(args.outfile, system_info=system_info)
+
+if args.platform is not None:
+    # Use specified subset of platforms
+    requested_platforms = args.platform.split(',')
+    if not set(requested_platforms).issubset(PLATFORMS):
+        parser.error(f'Available platforms: {PLATFORMS}')
+    PLATFORMS = requested_platforms
+
+if args.test is not None:
+    # Use specified subset of tests
+    requested_tests = args.test.split(',')
+    if not set(requested_tests).issubset(TESTS):
+        parser.error(f'Available tests: {TESTS}')
+    TESTS = requested_tests
+
+if args.precision is not None:
+    # Use specified subset of precisions
+    requested_precisions = args.precision.split(',')
+    if not set(requested_precisions).issubset(PRECISIONS):
+        parser.error(f'Available precisions: {PRECISIONS}')
+    PRECISIONS = requested_precisions
+else:
+    PRECISIONS = ['single'] # Peter's preferred default
```

Comment:
In this and the others below, why not use the `default` option to `add_argument()`?  Then this block could be simplified to

```python
precisions = args.precision.split(',')
if not set(precisions).issubset(PRECISIONS):
    parser.error(f'Available precisions: {PRECISIONS}')
```

Overwriting `PRECISIONS` is also confusing.  People will assume a name in all capitals is an immutable constant.  Putting the user's selection into a different variable called `precisions` is clearer.